dev/com.ibm.ws.cdi.extension_fat/fat/src/com/ibm/ws/cdi/extension/tests/WeldDevModeTest.java

```diff
@@ -0,0 +1,124 @@
+        multipleWarEmbeddedJar.addClass(EmbeddedJarMyEjb.class);
+        CDIArchiveHelper.addBeansXML(multipleWarEmbeddedJar, DiscoveryMode.ANNOTATED);
+
+        WebArchive multipleWarNoBeans = ShrinkWrap.create(WebArchive.class, "multipleWarNoBeans.war");
+        multipleWarNoBeans.addClass(NoBeansTestServlet.class);
+
+        WebArchive multipleWar = ShrinkWrap.create(WebArchive.class, "multipleWar1.war");
+        multipleWar.addClass(WAR1TestServlet.class);
+        multipleWar.addClass(WAR1MyBean.class);
+        multipleWar.addAsWebInfResource(WAR1TestServlet.class.getPackage(), "ejb-jar.xml", "ejb-jar.xml");
+        CDIArchiveHelper.addBeansXML(multipleWar, DiscoveryMode.ANNOTATED);
+        multipleWar.addAsLibrary(multipleWarEmbeddedJar);
+
+        EnterpriseArchive multipleWars = ShrinkWrap.create(EnterpriseArchive.class, "multipleWars2.ear");
+        multipleWars.setApplicationXML(NoBeansTestServlet.class.getPackage(), "application.xml");
+        multipleWars.addAsModule(multipleWar);
+        multipleWars.addAsModule(multipleWarNoBeans);
+
+        ShrinkHelper.exportDropinAppToServer(server, multipleWars, DeployOptions.SERVER_ONLY);
+
+        server.startServer(true);
+    }
+
+    @Test
+    public void testExtensionLoaded() throws Exception {
+        assertStringsInLogs("WELD-000411: Observer method \\[BackedAnnotatedMethod\\] public org.jboss.weld.probe.ProbeExtension.processAnnotatedTypes");
+        assertStringsInLogs("Weld Development Mode: ENABLED");
+
+        assertProbeProcessedBDA("multipleWars2#org.jboss.weld.probe.ProbeExtension.additionalClasses");
+        assertProbeProcessedBDA("multipleWars2#multipleWar1.war");
+        assertProbeProcessedBDA("multipleWars2#multipleWar1.war#WEB_INF_LIB#WEB-INF/lib/multipleWarEmbeddedJar.jar");
+        //assertProbeProcessedBDA("multipleWars2#com.ibm.ws.beanvalidation.v11.cdi_1.additionalClasses"); //BDA name is very different in EE9
+        //assertProbeProcessedBDA("multipleWars2#com.ibm.ws.jsf.2.2_1.additionalClasses");                //BDA name is very different in EE9
+        assertProbeProcessedBDA("multipleWars2#multipleWarNoBeans.war");
+        assertProbeProcessedBDA("multipleWars2#cdi.helloworld.extension.*_1.additionalClasses"); //BDA name varies slightly in EE9
+        assertProbeProcessedBDA("multipleWars2#io.openliberty.transaction.internal.cdi.*_1.additionalClasses");//BDA name varies slightly in EE9
+
+    }
+
+    private static void assertStringsInLogs(String msg) throws Exception {
+        Assert.assertFalse("Log Message Not Found: " + msg, server.findStringsInLogs(msg).isEmpty());
+    }
+
+    private static void assertProbeProcessedBDA(String bda) throws Exception {
+        assertStringsInLogs("PROBE-000002: Processing bean deployment archive: " + bda);
+    }
+
+    /**
+     * Post test processing.
+     *
+     * @throws Exception
+     */
+    @AfterClass
+    public static void cleanup() throws Exception {
```

Comment:
Can we move cleanup() so that it's next to setup()?

Reply:
Done